### ISSUE TITLE
`terragrunt plan / apply > tfplan.txt` does not pipe terraform plan to file tfplan.txt

### ISSUE THREAD

**Issue by hongshaoyang** [gruntwork-io/terragrunt#3667]
## Describe the bug

`terragrunt plan / apply > tfplan.txt` does not pipe terraform plan to file tfplan.txt

## Steps To Reproduce

In a working directory, run `terragrunt plan > tfplan.txt`. 

The following ouput appears:

```
$ terragrunt plan > now.txt
17:53:26.803 STDOUT terraform: data.huaweicloud_cce_cluster.this: Reading...
17:53:27.202 STDOUT terraform: data.huaweicloud_cce_cluster.this: Read complete after 0s [id=xxx]
17:53:27.305 STDOUT terraform: Terraform used the selected providers to generate the following execution
17:53:27.305 STDOUT terraform: plan. Resource actions are indicated with the following symbols:
17:53:27.305 STDOUT terraform:   + create
17:53:27.305 STDOUT terraform: Terraform will perform the following actions:
17:53:27.305 STDOUT terraform:   # kubernetes_namespace_v1.this will be created
17:53:27.305 STDOUT terraform:   + resource "kubernetes_namespace_v1" "this" {
17:53:27.305 STDOUT terraform:       + id                               = (known after apply)
17:53:27.305 STDOUT terraform:       + wait_for_default_service_account = false
17:53:27.305 STDOUT terraform:       + metadata {
17:53:27.305 STDOUT terraform:           + generation       = (known after apply)
17:53:27.305 STDOUT terraform:           + name             = "dne"
17:53:27.305 STDOUT terraform:           + resource_version = (known after apply)
17:53:27.305 STDOUT terraform:           + uid              = (known after apply)
17:53:27.305 STDOUT terraform:         }
17:53:27.305 STDOUT terraform:     }
17:53:27.305 STDOUT terraform: Plan: 1 to add, 0 to change, 0 to destroy.
17:53:27.305 STDOUT terraform: 
17:53:27.305 STDOUT terraform: 
17:53:27.305 STDOUT terraform: ─────────────────────────────────────────────────────────────────────────────
17:53:27.306 STDOUT terraform: Note: You didn't use the -out option to save this plan, so Terraform can't
17:53:27.306 STDOUT terraform: guarantee to take exactly these actions if you run "terraform apply" now.
```

And tfplan.txt is **an empty file** (!!)

## Expected behavior

stdout from terraform should be piped to file

## Nice to haves

- [ ] Terminal output
- [ ] Screenshots

## Versions

- Terragrunt version: terragrunt version 0.69.13
- OpenTofu/Terraform version: Terraform v1.10.2 on darwin_arm64
- Environment details (Ubuntu 20.04, Windows 10, etc.): macOS Ventura 13.6.7 (22G720)

**Comment by yhakbar** [comment]
Hey @hongshaoyang ,

Sorry that you're having this experience. What you're experiencing stems from how stdout/stderr is handled right now in Terragrunt. You can workaround this issue by using the `--terragrunt-forward-tf-stdout` flag.

e.g.

```bash
terragrunt plan --terragrunt-forward-tf-stdout > tfplan.txt
```

Because Terragrunt enriches OpenTofu/Terraform logging by default, right now the stdout stream from OpenTofu/Terraform is being emitted in the stderr stream of Terragrunt (with the rest of logging).

We had a chat about this, and it's not the right experience for new users (or long time users). We're changing how this works. 

I've asked @levkohimins to look into changing it so that we split logging into two streams, for stdout and stderr, with all STDOUT level logging emitted to Terragrunt stdout.

In the future, the command that you ran above will work by default, without using any extra flags.

**Comment by yhakbar** [comment]
Released in [v0.71.0](https://github.com/gruntwork-io/terragrunt/releases/tag/v0.71.0).
